### ISSUE TITLE
Vim9: noclear does not preserve autoload import items across reloads

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#9593]
**Steps to reproduce**

Run this shell command:

    vim -Nu NONE -S <(tee <<'EOF'
        vim9script
        var dir = '/tmp/.vim'
        dir->delete('rf')
        &runtimepath = dir
        dir ..= '/autoload'
        dir->mkdir('p')
        var lines =<< trim END
            vim9script
            export def Func()
                echomsg 'from Func()'
            enddef
        END
        lines->writefile(dir .. '/script.vim')
        lines =<< trim END
            vim9script noclear
            if exists('g:loaded')
                finish
            endif
            g:loaded = true
            import autoload 'script.vim'
            nnoremap <F3> <ScriptCmd>script.Func()<CR>
        END
        lines->writefile('/tmp/t.vim')
        source /tmp/t.vim
        source /tmp/t.vim
        feedkeys("\<F3>")
    EOF
    )

`E1048` is given:

    E1048: Item not found in script: Func

**Expected behavior**

No error is given, and `from Func()` is echo'ed.  Because `noclear` has preserved the `script` variable *and* the `Func` item.  Not just the `script` variable.

**Version of Vim**

8.2 Included patches: 1-4174

**Environment**

Operating system: Ubuntu 20.04.3 LTS
Terminal: xterm
Value of $TERM: xterm-256color
Shell: zsh 5.8

**Additional context**

If `noclear` is removed, the error becomes:

    E121: Undefined variable: script

Test:
```vim
vim9script
var dir = '/tmp/.vim'
dir->delete('rf')
&runtimepath = dir
dir ..= '/autoload'
dir->mkdir('p')
var lines =<< trim END
    vim9script
    export def Func()
        echomsg 'from Func()'
    enddef
END
lines->writefile(dir .. '/script.vim')
lines =<< trim END
    vim9script
    if exists('g:loaded')
        finish
    endif
    g:loaded = true
    import autoload 'script.vim'
    nnoremap <F3> <ScriptCmd>script.Func()<CR>
END
lines->writefile('/tmp/t.vim')
source /tmp/t.vim
source /tmp/t.vim
feedkeys("\<F3>")
```
    E121: Undefined variable: script

This error is correct and shows that `noclear` has an effect.

---

If `script.vim` is moved from `autoload/` to `import/`, the issue disappears:
```vim
vim9script
var dir = '/tmp/.vim'
dir->delete('rf')
&runtimepath = dir
dir ..= '/import'
dir->mkdir('p')
var lines =<< trim END
    vim9script
    export def Func()
        echomsg 'from Func()'
    enddef
END
lines->writefile(dir .. '/script.vim')
lines =<< trim END
    vim9script noclear
    if exists('g:loaded')
        finish
    endif
    g:loaded = true
    import 'script.vim'
    nnoremap <F3> <ScriptCmd>script.Func()<CR>
END
lines->writefile('/tmp/t.vim')
source /tmp/t.vim
source /tmp/t.vim
feedkeys("\<F3>")
```
    from Func()

Why does `noclear` preserve imported items when a script is reloaded under `import/`, but not when it's reloaded under `autoload/`?  Is it because of patch [8.2.1398](https://github.com/vim/vim/releases/tag/v8.2.1398)?

**Comment by brammool** [comment]
Interestingly, it works when <F3> is used in between the two times the script is sourced.

It happens because the "vim9script" marks the import as "not cleared", which changes the flag that also indicates it's an autoload script.

**Comment by brammool** [comment]
Fixed by patch  8.2.4177
